### ISSUE TITLE
masking/usability proposals

### ISSUE THREAD

**Issue by Heddesdorfer** [hanatos/vkdt#102]
Preamble
I don't know, if all of this points are really new, or if I just didn't have seen how to do...nevertheless 

1. If you want to use masking, you have to add more than one module. It would be nice to be able to choose more than module in the dialog before it will be closed.
2. If you want to mask another part, it would be helpful if you could copy and past the modules in the node editor, preserving inner and outer connections, but without changed parameters.
3. 'undo masking' is also a nice to have.
4. If I use the standard cfg, some names like 'display: main' or 'display:hist' are set, but I see no possibility to give modules a meaningful name.

**Comment by hanatos** [comment]
thanks for these observations. i assume you found the related links in the docs?

https://vkdt.org/doc/howto/draw-masks/readme.html
https://vkdt.org/src/pipe/modules/draw/readme.html
https://vkdt.org/src/pipe/modules/mask/readme.html

i think they are slightly outdated now, i should update them. how are you masking? `draw`, `mask` or `grad` module?

1. yis. i think this should be realised via presets (`ctrl-p` should show `dodge` or `burn` as it does for `inpaint.pst` and `retouch.pst`) the advantage would be that the connections etc are useful out of the box, too.
2. you can add another stroke in the `draw` module..
3. ..and remove it by right clicking, if that is what you mean? of course there's the `ctrl-h` history stack in the left panel too.
4. right. you can't rename instances. is that a useful feature? you can name the module when adding it (the first textbox in the `ctrl-m` popup, "instance"). changing it after the fact seemed not very interesting to me (prove me wrong!) and it's a delicate thing to do in the processing history.

from the discussion so far i see the following action items:
* update mask/draw docs
* introduce copy/paste node selection in graph editor
* introduce presets for common masked edits (dodge/burn, what else?)

**Comment by Heddesdorfer** [comment]
Yes, I found the links - and because I'm impatient I tried the draw-guided.cfg from the example.
ad 2: but then both strokes would be handled in the same way, what if the second one needs e.g. an other exposure?
ad 3: ah! I must have overseen these control-thingies description.
ad 4: I didn't realize that "instance" is used for naming - that's sufficient.

**Comment by Heddesdorfer** [comment]
I just tried dodge & burn presets, works well.
But the modules are strangely listed in darkroom mode:
![modules](https://github.com/hanatos/vkdt/assets/91422232/4bee387d-ab0b-43b8-9d93-c161cf4c5264)

If I need another instance of dodge / burn (because of another degree of exposure), the preset doesn't work.
But that might be a feature beyond raw processing.

**Comment by hanatos** [comment]
the module sorting is just a tree traversal, loosely expressing the
dependencies of the modules (same as darktable, closer to output is near
the top). i suppose i could sort by instance name within these constraints.

right, can't apply more than one instance of the same called `dodge`. you
found out how to change opacity for individual brush strokes?
(shift-scrollwheel) one strategy for burning would be setting the exposure
to the max required and then lower the opacity/hardness/size of individual
brush strokes as required.

in fact i think i want to make brush strokes carry rgb information too (to
be used as more input parameters for blend modules of the future, or as
light painting of some kind).

note to self: i also want versions of the dodge/burn presets with a guided
filter wired to stick to edges more.

**Comment by hanatos** [comment]
pushed the guided fillter draw presets.
also now you can filter by module name and module instance:
[image: 2024-01-24-200137_593x408_scrot.png]

**Comment by Heddesdorfer** [comment]
Works!
What's now ...irritating... when I start vkdt, it starts in full screen mode. What means, I have to press F11 to move it to another monitor, where I can enlarge it. I know, that's not really an error, but if something changes "suddenly"...

I work on my notebook, and programs start on this monitor. If available, I use an external one, so I have to move vktd. So it may be a my personal inconvenience.

**Comment by hanatos** [comment]
pushed something about the f11 and window decoration thing. it should now
start unfullscreened, which i find very inconvenient indeed. i might make
this a config option.
